lib/Current.dart

```diff
@@ -0,0 +1,706 @@
+        Expanded(
+          flex: 1,
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.start,
+            mainAxisSize: MainAxisSize.max,
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: <Widget>[
+              ClipRRect(
+                borderRadius: BorderRadius.circular(5.0),
+                child: Container(
+                  width: 18,
+                  height: 18,
+                  decoration: new BoxDecoration(
+                    color: Colors.grey,
+                    shape: BoxShape.circle,
+                    border: new Border.all(
+                        color: Colors.white,
+                        width: 6.0,
+                        style: BorderStyle.solid),
+                    boxShadow: [
+                      new BoxShadow(
+                        color: Colors.grey[300],
+                        blurRadius: 30.0,
+                      )
+                    ],
+                  ),
+                  child: Text(
+                    "",
+                  ),
+                ),
+              ),
+              Container(
+                width: 3,
+                height: 20,
+                decoration: new BoxDecoration(
+                  color: Colors.transparent,
+                  shape: BoxShape.rectangle,
+                ),
+                child: Text(""),
+              ),
+            ],
+          ),
+        ),
+        Expanded(
+          flex: 9,
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            mainAxisSize: MainAxisSize.max,
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: <Widget>[
+              Text('$title\n $subTile',
+                  style: TextStyle(
+                      fontFamily: "regular",
+                      fontSize: 19,
+                      color: Colors.black54)),
+            ],
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class RepeatRideButton extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialButton(
+      padding: EdgeInsets.only(
+        bottom: SizeConfig.safeBlockHorizontal * 7,
+      ),
+      color: Colors.white,
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: <Widget>[
+          Icon(Icons.repeat),
+          SizedBox(width: 10),
+          Text(
+            'Repeat This Ride',
+            textAlign: TextAlign.start,
+            style: TextStyle(
+                color: Colors.black,
+                fontFamily: "SFPro",
+                fontSize: 20,
+                fontWeight: FontWeight.bold),
+          )
+        ],
+      ),
+      onPressed: () {},
+    );
+  }
+}
+
+class RideStatus extends StatefulWidget {
+  @override
+  _RideStatusState createState() => _RideStatusState();
+}
+
+class _RideStatusState extends State<RideStatus> {
+  @override
+  Widget build(BuildContext context) {
```

Comment:
Refactor the copy-pasted parts of this build function into separate widgets that can be re-used. Refactor as much as possible. If you used it more than once or if it's hard to read, I recommend factoring it out into its own widget.